### PR TITLE
Make marketing announcements translatable

### PR DESCRIPTION
**Pull request by rshipp**
Adds a `get_localized_announcement_for_page` method in the `Announcements` class, and uses it to expose translatable strings in the homepage. This does not update other pages that use the marketing banner to use the localized strings, though that could be part of future work.

Since the banners are now translatable, we want to always show the banner, instead of restricting it to only english-locale users.

At Mario's suggestion, this proposal moves the marketing `announcements.json` out of pegasus and into dashboard. I put it in `dashboard/config/marketing/announcements.json`, with i18n locales expected in `dashboard/config/locales/marketing_announcements-*.json`, but this name/structure can be easily changed if someone wants something different.

Thank you to Mario, Kelby, and Molly for providing some early suggestions and feedback on this approach.

## Links

- jira ticket: [ACQ-946](https://codedotorg.atlassian.net/browse/ACQ-946)

## Testing story

* Verified `localhost.code.org:3000` and the dashboard teacher homepage still load and showed the appropriate announcements. 
* Verified an i18n file with the expected format is correctly used to supply translated strings, when available.
* Updated a unit test, and verified all existing tests pass.

Note the below screenshot uses an arbitrary test string, not an actual translation of the announcement.

Screenshot 2023-09-27 124649-marketing-i18n

### CODE REVIEW

**Review by TurnerRiley — APPROVED** [review]
Thank you for doing this Ryan, this looks great!

File: dashboard/config/marketing/announcements.json
Comment (davidsbailey) [inline dashboard/config/marketing/announcements.json:1]: @kelbyhawn are you the only person that needs to edit this file now on the marketing side? just calling out that this won't be accessible to other marketing folks via dropbox any more

**Review by davidsbailey — APPROVED** [review]
Looks good to me!

File: dashboard/config/marketing/announcements.json
Comment (kelbyhawn) [reply dashboard/config/marketing/announcements.json:1]: This is ok 👍 We've been moving away from having non-devs make updates.

File: dashboard/lib/announcements.rb [hunk 1]
```diff
@@ -24,6 +24,35 @@ def self.get_announcement_for_page(page)
...
+      buttonText: I18n.t(
+        "buttonText",
+        default: announcement['buttonText'],
+        scope: [:data, :marketing_announcements, :banners, announcement[:id]],
+        smart: true
+      ),
+    }
+
+    announcement.merge(localized_fields)
+  end
```
Comment (mgc1194) [inline dashboard/lib/announcements.rb]:
```suggestion
  # gets localized special announcement data for a page, or nil if not found
  def self.get_localized_announcement_for_page(page)
    announcement = get_announcement_for_page(page)
    return nil unless announcement
    announcement_scope = [:data, :marketing_announcements, :banners, announcement[:id]]

    localized_fields = {
      title: localize_property("title", announcement['title'],announcement_scope),
      body: localize_property("body", announcement['body'], announcement_scope),
      buttonText: localize_property( "buttonText", announcement['buttonText'], announcement_scope),
    }

    announcement.merge(localized_fields)
  end
  
  def localize_property(property_name, default, scope)
    I18n.t(
        property_name,
        default: default,
        scope: scope,
        smart: true
      )
  end
```

Comment (rshipp) [reply dashboard/lib/announcements.rb]: good call! will refactor this out

**Pull request merged**